### ISSUE TITLE
@types/mocha ITestCallbackContext.slow() should return this

### ISSUE THREAD

**Issue by thynson** [DefinitelyTyped/DefinitelyTyped#15028]
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @kazimanzurrashid @otiai10 @jt000 @enlight

`ITestCallbackContext.slow()` should return `this`, otherwise the following code is not passed.
```typescript
describe('some test', function() {
    it('should pass', function() {
        this.slow(5000).timeout(10000); // error TS2339: Property 'timeout' does not exist on type 'void'.
        assert(true);
    });
});
```
All methods of `ITestCallbackContext` should return `this`, as well as `ISuiteCallbackContext`, and `IHookCallbackContext`.

**Comment by thynson** [comment]
Here is a simple test file runs by mocha3
```javascript
describe('hook', function() {
	this.retries(3).timeout(2000).slow(1000).retries(3);
	beforeEach(function() {
		this.skip().timeout(2000).skip();
	});
});

describe('test', function() {
	let n = 0;
	it('should pending', function() {
		this.retries(3).timeout(1000).slow(500).skip().slow(500);
	})
	it('should pending', function() {
		this.retries(3);
		if (n++ == 0) throw new Error();
	})

})
```

**Comment by thynson** [comment]
Make pull request #15031

**Comment by thynson** [comment]
Closed for #15031 have been merged
